### PR TITLE
julia_111{,-bin}: 1.11.1 -> 1.11.2, julia_19{,-bin}: drop

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NickCao** [comment]
@ofborg build julia_111 julia_111-bin

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -572,6 +572,8 @@ mapAliases {
 
   # Julia
   julia_16-bin = throw "'julia_16-bin' has been removed from nixpkgs as it has reached end of life"; # Added 2024-10-08
+  julia_19-bin = throw "'julia_19-bin' has been removed from nixpkgs as it has reached end of life"; # Added 2024-12-02
```
Comment (thomasjm) [inline pkgs/top-level/aliases.nix:575]: AFAICT Julia 1.9 is still maintained, do we actually need to remove it at this point?

It sounds like 1.6 has become unmaintained, but releases after that are still maintained:

https://discourse.julialang.org/t/julia-v1-11-0-has-been-released-and-v1-10-is-now-lts/121064

**Review by thomasjm — COMMENTED** [review]
`julia_111` looks good, with 8 package failures out of 1000.

`julia_111-bin` is failing on all package builds. Actually, it's also failing on `master` with the same issue. The main error seems to be this:

```
> nix run --impure --expr 'with import ./. {}; julia_111-bin.withPackages ["Plots"]' -L
...
LoadError: GitError(Code:ERROR, Class:SSL, failed to load CA certificates: 0xffffc200 - PK - Read/write of file failed)
...
ERROR: Failed to precompile Pkg [44cfe95a-1eb2-52ea-b672-e2afdf69b78f] to "/build/home/.julia/compiled/v1.11/Pkg/jl_z0P6VE".
```

It probably doesn't need to block this PR, but `Pkg`-related stuff in `julia_111-bin` seems to have big problems at the moment (cf also https://github.com/NixOS/nixpkgs/issues/350590).

File: pkgs/top-level/aliases.nix (hunk 1 above)
Comment (NickCao) [reply pkgs/top-level/aliases.nix:575]: It that case let's leave it around for a while more.

**Review by thomasjm — APPROVED** [review]
LGTM, thanks!

I do still see these errors that we had with `julia_111-bin` before, any idea what it's about?

```shell
[nix-shell:~/.cache/nixpkgs-review/pr-361178/nixpkgs]$ nix run .#julia_111-bin
warning: Git tree '/home/tom/.cache/nixpkgs-review/pr-361178/nixpkgs' is dirty
[ Info: Precompiling InteractiveUtils [b77e0a4c-d291-57a0-90e8-8db25a27a240] (cache misses: ocachefile invalid checksum (1), mismatched flags (1))
[ Info: Precompiling REPL [3fa0cd96-eef1-5676-8a61-b3b8758bbffb] (cache misses: wrong dep version loaded (1), mismatched flags (1))
```

**Comment by NickCao** [comment]
> I do still see these errors that we had with `julia_111-bin` before, any idea what it's about?

Not yet, I will investigate.

**Comment by vbgl** [comment]
Can we backport this to 24.11?

**Pull request merged**